### DISCUSSION TITLE
la cuisine à données – propluvia-cartes

### DISCUSSION THREAD

**Discussion by giscus** [taniki/data#3] [General]
# la cuisine à données – propluvia-cartes

https://data.11d.im/foodcourt/crise-climatique/propluvia-cartes.html

**Comment by CoquelicEAU** [comment]
Bonjour, 
Je vois que vous récupérez les données via "https://raw.githubusercontent.com/taniki/propluvia/main/propluvia.csv" est ce un autre import avec l'entièreté des données par rapport à "http://data.cquest.org/propluvia/"? 
 
Je ne comprends pas comment vous associez les codes Za (et avec quel fichier carto) avec avec leur localisation sur la carte, c'est probablement ici : 

```
.set_index('Code_ZA')
.join(zones)
.query('not codeDep.isin(@zones_exclude)')
```


J'aimerais construire un fichier shp (pour Qgis) avec toutes les localisations et sommes des jours d'arrêté par type (crise, alerte etc) et par année. Comment puis-je procéder pour récupérer ce set de donné ?

**Reply by taniki** [reply]
Le fichier `csv` que vous pointez est une concaténation des fichiers `csv` de `data.cquest.org`.

le fichier [SHP_20220806.geojson](https://data.cquest.org/propluvia/SHP_20220806.geojson) contient les geometry des bassins versants. La clé de jointure est `codeZA` côté shapefile et `Code_ZA` côté csv des arrêtés. Je prépare la jointure en mettant les deux colonnes en index de chaque côté.

Pour compter le nombre de jours par année et par arrêté, il faut quelques calculs car les débuts et fins ne sont pas toujours la même année et parfois pas défini. J'ai un bout de code pour faire ça en pandas qui n'est pas encore en ligne.

**Reply by taniki** [reply]
voici un fichier avec les arrêtés, le code ZA concerné, leur niveau ainsi que le nombre de jour pour chaque année https://gist.githubusercontent.com/taniki/dce34cdaba0f08c30c9b23f637b3b036/raw/d32f34107564e7479b2da3905f19cff0e6534ab6/arretes_zones_annees_jours.csv
